src/Frontend/TableHelper/Animation.js

```diff
@@ -24,227 +19,258 @@ export const stopStatus = { // 暫停時的狀態
 
-            }
+        this.getItems(path, pathDirection)
+
+        this.id = this.items[0]
+        if(this.id === undefined || path.length === 0){
+            return false
         }
-        count += 1
-    }, speed)
+        
+        this.newid = path[0]
+        this.d = pathDirection[0]
+
+        this.getClassName(this.newid, this.d)
+        setTable((WhichComponentSame(this.id) >= 3) ? this.id : -1, this.className1)
+        setTable(this.newid, this.className2)
+        return false
+    }
+    
 }
 
-export function PathAnimation(path, speed, pathDirection, sysStatusFunction, updateFunction) {
-    var [id, newid] = [stopStatus.pathID[0], stopStatus.pathID[1]]
-    var count = stopStatus.path
+function getAnimationObj(){
+    var animationObj
+    if(stopStatus.search.length > 0){
+        animationObj = [new searchAnimation_search(), stopStatus.search];
+    } else if (stopStatus.bomb.length > 0){
+        animationObj = [new searchAnimation_bomb(), stopStatus.bomb];
+    } else if (stopStatus.path.length > 0){
+        animationObj = [new searchAnimation_path(), [stopStatus.path, stopStatus.pathDirection]];
+    } else {
+        animationObj = [undefined, undefined];
+    }
+    return animationObj
+}
 
-    const pathAnimation = setInterval(() => {
-        if (count === path.length) {
-            resetAnimation()
-            updateFunction()
-            clearInterval(pathAnimation)
+export function SearchAnimation(search, bomb, path, pathDirection, speed, sysStatusFunction, updateFunction) {
+    if(stopStatus.complete){
+        path.unshift(-1)
+        pathDirection.unshift("")
+    
+        // I don't know why ???
+        // setSearchAnimation(search, path, pathDirection, bomb, "")
+        stopStatus.search = search
+        stopStatus.path = path
+        stopStatus.pathDirection = pathDirection
+        stopStatus.bomb = bomb
+        stopStatus.complete = false
+    } 
 
-        }else {
-            if(stopStatus.animationStatus === false){
-                stopStatus.path = count
-                stopStatus.pathID = [id, newid]
-                sysStatusFunction()
-                clearInterval(pathAnimation)
-                return
+    var items, animationObj, positions;
 
-            }else{
-                if (WhichComponentSame(path[count]) > 3) {
-                    newid = path[count]
-                    setTable(id, componentKind.path)
-                    setTable(newid, direction(pathDirection[count]))
-                    id = newid
-                }else{
-                    setTable(id, componentKind.path)
-                    setTable(path[count], StartEndBombWeight(WhichComponentSame(path[count]), direction(pathDirection[count]), direction(pathDirection[count]), componentKind.bombPath, componentKind.weightPath))
-                }
+    const searchAnimation = setInterval(() => {
+        items = getAnimationObj()
+        animationObj = items[0]
+        positions = items[1]
```

Comment:
"animationObj" and "positions" might be "undefined". 
Please handle this scenario in case some goes unexpectedly.

Reply:
if animationObj equal to "undefined", function will break (clearInterval line 167 ~ 169)

Reply:
Please handle "positions" as well.
`if (animationObj === undefined || positions ==== undefined) {`